plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -678,41 +678,74 @@ private Optional<String> formatProperty(List<String> prop)
         };
     }
 
     private static LongWriteFunction uInt16WriteFunction()
     {
-        return (statement, index, value) -> {
-            // ClickHouse stores incorrect results when the values are out of supported range.
-            if (value < UINT16_MIN_VALUE || value > UINT16_MAX_VALUE) {
-                throw new TrinoException(INVALID_ARGUMENTS, format("Value must be between %s and %s in ClickHouse: %s", UINT16_MIN_VALUE, UINT16_MAX_VALUE, value));
+        return new LongWriteFunction()
+        {
+            @Override
+            public void setNull(PreparedStatement statement, int index) throws SQLException
+            {
+                statement.setNull(index, Types.INTEGER);
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, long value) throws SQLException
+            {
+                // ClickHouse stores incorrect results when the values are out of supported range.
+                if (value < UINT16_MIN_VALUE || value > UINT16_MAX_VALUE) {
+                    throw new TrinoException(INVALID_ARGUMENTS, format("Value must be between %s and %s in ClickHouse: %s", UINT16_MIN_VALUE, UINT16_MAX_VALUE, value));
+                }
+                statement.setInt(index, toIntExact(value));
             }
-            statement.setInt(index, toIntExact(value));
         };
     }
 
     private static LongWriteFunction uInt32WriteFunction()
     {
-        return (preparedStatement, parameterIndex, value) -> {
-            // ClickHouse stores incorrect results when the values are out of supported range.
-            if (value < UINT32_MIN_VALUE || value > UINT32_MAX_VALUE) {
-                throw new TrinoException(INVALID_ARGUMENTS, format("Value must be between %s and %s in ClickHouse: %s", UINT32_MIN_VALUE, UINT32_MAX_VALUE, value));
+        return new LongWriteFunction()
+        {
+            @Override
+            public void setNull(PreparedStatement statement, int index) throws SQLException
+            {
+                statement.setNull(index, Types.BIGINT);
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, long value) throws SQLException
+            {
+                // ClickHouse stores incorrect results when the values are out of supported range.
+                if (value < UINT32_MIN_VALUE || value > UINT32_MAX_VALUE) {
+                    throw new TrinoException(INVALID_ARGUMENTS, format("Value must be between %s and %s in ClickHouse: %s", UINT32_MIN_VALUE, UINT32_MAX_VALUE, value));
+                }
+                statement.setLong(index, value);
             }
-            preparedStatement.setLong(parameterIndex, value);
         };
     }
 
     private static ObjectWriteFunction uInt64WriteFunction()
     {
         return ObjectWriteFunction.of(
                 Int128.class,
+                2, // Taken from JdbcTypeMapping class
```

Comment:
nit: what type does `2` signify? Might be worthwhile to add it to the comment (I'm guessing Int?)